modules/postprocessing/src/processors/depthoffield.cpp

```diff
@@ -0,0 +1,427 @@
+}
+
+double DepthOfField::calculateFocusDepth() {
+    if (manualFocus_ || !trackingInport_.hasData()) {
+        return focusDepth_.get();
+    } else {
+        vec3 lookFrom = ogCamera_->getLookFrom();
+        vec3 lookTo = ogCamera_->getLookTo();
+        vec3 front = glm::normalize(lookTo - lookFrom);
+
+        auto mesh = trackingInport_.getData();
+        const vec3 focusPoint = mesh->getBuffer(0)->getRepresentation<BufferRAM>()->getAsDVec3(0);
+        return glm::length(glm::dot(focusPoint - lookFrom, front) * front);
+    }
+}
+
+vec2 DepthOfField::calculatePeripheralCameraPos(int evalCount, int maxEvalCount) {
+    if (evalCount == 0) {
+        return vec2(0, 0);
+    } else {
+        float currAngle = 2.0 * M_PI * (float(evalCount) - 0.5) / float(maxEvalCount - 1);
+        return aperture_.get() / 2.0 * vec2(glm::cos(currAngle), glm::sin(currAngle));
+    }
+}
+
+void DepthOfField::addToAccumulationBuffer(std::shared_ptr<const Image> img,
+                                           TextureUnitContainer& cont) {
+    utilgl::activateTargetAndCopySource(*nextOutImg_, inport_, ImageType::ColorOnly);
+    addSampleShader_.activate();
+
+    img->getRepresentation<ImageGL>();
+    prevOutImg_->getRepresentation<ImageGL>();
+    utilgl::bindAndSetUniforms(addSampleShader_, cont, *img, "newImg", ImageType::ColorOnly);
+    utilgl::bindAndSetUniforms(addSampleShader_, cont, *prevOutImg_, "oldImg",
+                               ImageType::ColorOnly);
+    addSampleShader_.setUniform("nOldImages", evalCount_);
+
+    utilgl::singleDrawImagePlaneRect();
+    addSampleShader_.deactivate();
+}
+
+void DepthOfField::warpToLightfieldGPU(TextureUnitContainer& cont, double nearClip, double farClip,
+                                       double fovy, double focusDepth, size2_t dim,
+                                       vec2 cameraPos) {
+    glActiveTexture(GL_TEXTURE);
+    auto lightFieldGL = lightField_->getEditableRepresentation<VolumeGL>();
+    auto lightFieldTexHandle = lightFieldGL->getTexture()->getID();
+    glBindImageTexture(0, lightFieldTexHandle, 0, GL_FALSE, 0, GL_WRITE_ONLY, GL_RGBA32F);
+
+    auto lightFieldDepthGL = lightFieldDepth_->getEditableRepresentation<VolumeGL>();
+    auto lightFieldDepthTexHandle = lightFieldDepthGL->getTexture()->getID();
+    glBindImageTexture(1, lightFieldDepthTexHandle, 0, GL_FALSE, 0, GL_READ_WRITE, GL_R32F);
+
+    addToLightFieldShader_.activate();
+    utilgl::bindAndSetUniforms(addToLightFieldShader_, cont, inport_, ImageType::ColorDepth);
+    utilgl::bindAndSetUniforms(addToLightFieldShader_, cont, *haltonImg_, "halton",
+                               ImageType::ColorOnly);
+
+    addToLightFieldShader_.setUniform("nearClip", float(nearClip));
+    addToLightFieldShader_.setUniform("farClip", float(farClip));
+    addToLightFieldShader_.setUniform("evalCount", int(evalCount_));
+    addToLightFieldShader_.setUniform("aperture", float(aperture_.get()));
```

Comment:
I'm a little bit unsure about making this change since I don't currently have access to a machine where I can test the compute shader. Additionally, as far as I can tell, this can only be used on properties/ports which have identifiers so the majority of these lines would be kept the same anyway.

Reply:
Ok. 
Setting the camera property would also set the near/far
https://github.com/inviwo/inviwo/blob/2205ee567b625a65c8aeccb78cd56ea15d5c4e6c/modules/opengl/glsl/utils/structs.glsl

Reply:
Hmm yeah I see how that would be a better design choice... Is it acceptable the way it is though? As I've mentioned, I don't have access to a machine to test the compute shader and I'm worried I might break something. (I was only supposed to be working on this project until the end of june but there have been various delays).